h2/src/main/org/h2/table/JoinBatch.java

```diff
@@ -1074,8 +1074,8 @@ protected void startQueryRunners(int resultSize) {
      * Query runner for UNION.
      */
     private static class QueryRunnerUnion extends QueryRunnerBase {
-        Future<Cursor>[] topFutureCursors;
-        private ViewIndexLookupBatchUnion batchUnion;
+        private final Future<Cursor>[] topFutureCursors;
```

Comment:
Please remove `private` modifier from here. This field is accessed from another class, Java language allows such access, but JVM does not, this leads to generation of synthetic method by Java compiler.

Other changes are reasonable from my point of view.